backend/src/nodes/nodes/rest/img2img.py

```diff
@@ -0,0 +1,99 @@
+from __future__ import annotations
+
+import numpy as np
+
+from . import category as RESTCategory
+from ...impl.rest import (
+    decode_base64_image,
+    SamplerName,
+    STABLE_DIFFUSION_IMG2IMG_URL,
+    post_async,
+    encode_base64_image,
+)
+from ...node_base import AsyncNodeBase
+from ...node_factory import NodeFactory
+from ...properties.inputs import (
+    TextInput,
+    NumberInput,
+    EnumInput,
+    ImageInput,
+)
+from ...properties.outputs import LargeImageOutput
+
+
+@NodeFactory.register("chainner:rest:sd_img2img")
+class Img2Img(AsyncNodeBase):
+    def __init__(self):
+        super().__init__()
+        self.description = "Modify an image using an external Stable Diffusion service"
+        self.inputs = [
+            ImageInput(),
+            TextInput("Prompt", default="an astronaut riding a horse"),
+            TextInput("Negative Prompt", default=""),
+            NumberInput(
+                "Denoising Strength",
+                minimum=0,
+                default=0.75,
+                maximum=1,
+                controls_step=0.1,
+                precision=4,
+            ),
+            NumberInput("Seed", minimum=0, default=42, maximum=4294967296),
+            NumberInput("Steps", minimum=1, default=20, maximum=150),
+            EnumInput(SamplerName, default_value=SamplerName.EULER),
+            NumberInput(
+                "CFG Scale",
+                minimum=1,
+                default=7,
+                maximum=20,
+                precision=4,
+                controls_step=0.1,
+            ),
+            NumberInput("Width", minimum=64, default=512, maximum=2048),
+            NumberInput("Height", minimum=64, default=512, maximum=2048),
```

Comment:
These are the output width and height of the image below, right? Does the REST service guarantee that it will return images with example those dimensions?

Can these be any number between 64 and 2048 or are only powers of 2 allowed? I remember playing with an SD upscaler, and it only correctly support multiples of 64.

Reply:
Yes.  I don't know if it explicitly guarantees the output size, but it does seem to either produce output of the given size or crash.  It looks like it works with any integer values, though I believe models work best on the dimensions they were trained on.  (Usually 512x512 or 768x768.)

In the Automatic UI, these are sliders from 64 to 2048.  I used another UI that had a dropdown menu for these with various canonical values.

Imitating the UI decisions of the thing we're calling seems like a good baseline.